src/test/cypress/README.md

```diff
@@ -1,8 +1,112 @@
+
+A user with the `ADMIN` role is needed to create and delete courses. That account does not have a fixed `ID` because the template is not used for that user. The credentials for the `ADMIN` account are provided separately in the configuration of the test suite.
+
+:warning: **The guided tour should be disabled for every user required by the test suite. Otherwise some/all tests in the suite will fail!**
+
+## Test suite configuration
+### General configuration
+Once the test environment is running and the required users are added in the test environment the test suite needs to be configured. The configuration is done via two files:
+1. `cypress.json`: Contains general settings for Cypress
+2. `cypress.env.json`: Contains settings specific for Artemis
+
+In the following we will explain what setting in the configuration files has to be adjusted to be able to execute the test suite:
+For `cypress.json`:
+* `baseUrl`: The url pointing to the test environment here (make sure that there is no trailing slash)
+
+For `cypress.env.json`:
+* `username`: The username template with the `USERID` here (as described in the [pre-created users section](#pre-created-users))
+* `password`: The password template
+* `adminUsername`: The admin username (no template)
+* `adminPassword`: The admin password (no template)
+
+### Additional configuration for test environments using the Bamboo + Bitbucket setup
+On Bamboo + Bitbucket setups Artemis has issues with the group synchronization if a new course is created and immediately afterwards a programming exercise is created. This requires the tests to wait for over one minute in each test spec, which is related to programming exercises. This increases the total execution time of the test suite by a lot. The issue can be avoided by using already existing user groups in the course creation.
+
+Currently the tests require one pre-created user group for the roles `USER`, `TEACHING ASSISTANT`, `EDITOR` and `INSTRUCTOR` in the user management system of the test environment. Each group has its own setting in the configuration file. The `allowGroupCustomization` setting has to be set to `true` otherwise the tests will not use the pre-created user groups (resulting in a failure of all programming exercise related tests on a Bamboo + Bitbucket setup).
+* `allowGroupCustomization`: `true`
+* `studentGroupName`: The group name for students (e.g. `artemis-e2etest-students`)
+* `tutorGroupName`: The group name for tutors (e.g. `artemis-e2etest-tutors`)
+* `editorGroupName`: The group name for tutors (e.g. `artemis-e2etest-editors`)
+* `instructorGroupName`: The group name for tutors (e.g. `artemis-e2etest-instructors`)
+
+### Example configurations
+In the following we will show example configurations of the test suite for imaginary Artemis setups. For readability we will leave out default settings, which are not mandatory to be adjusted.
```

Comment:
There are several settings in the configuration files, which contain default values to adjust Cypress for Artemis. These settings don't really have to be changed by the user. For example the `cypress.json` file contains settings which define where specific files can be found in the project (files containing tests, plugins, fixtures,...) or settings which define the default viewport dimensions the tests should use.
Those settings are all Cypress related settings, which don't need to be changed, so I left them out so that the user can focus on the settings, which actually require them to adjust something.
I also linked to the general Cypress configuration at the end in case users still want to know more about those settings.

Reply:
I changed the phrasing a bit.
But maybe it makes sense to not omit the unchaged values? This way the readers could copy and paste the example configurations to their local files and adjust what they need? This only would affect the `cypress.json` file anyways, because the every setting in the `cypress.env.json` file requires a change by the user.

Reply:
Here is how the `cypress.json` file for a imaginary setup would look like:
``` json
{
    "fixturesFolder": "fixtures",
    "integrationFolder": "integration",
    "pluginsFile": "plugins/index.ts",
    "screenshotsFolder": "screenshots",
    "videosFolder": "videos",
    "supportFile": "support/index.ts",
    "baseUrl": "https://imaginary-artemis-server.com",
    "video": false,
    "screenshotOnRunFailure": false,
    "viewportWidth": 1920,
    "viewportHeight": 1080,
    "defaultCommandTimeout": 20000,
    "responseTimeout": 120000
}
```
Except for the `baseUrl` property nothing changed from the configuration provided by us on Github, so I left all the other settings out